contracts/oracle/EdgeDataStreamVerifier.sol

```diff
@@ -0,0 +1,193 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "../utils/Cast.sol";
+import "../error/Errors.sol";
+
+contract EdgeDataStreamVerifier {
+
+    struct Report {
+        bytes32 feedId;
+        uint256 bid; // bid: min price, highest buy price
+        uint256 ask; // ask: max price, lowest sell price
+        uint32 timestamp;
+        int32 expo; // precision of bid&ask (negative value)
+    }
+
+    address public immutable TRUSTED_SIGNER;
```

Comment:
in other contracts, camel case is used for immutable variables, i think camel case should be used here as well